spdx/writers/jsonyamlxml.py

```diff
@@ -533,34 +533,45 @@ def create_ext_document_references(self):
 
         return ext_document_reference_objects
 
-    def create_relationships(self) -> Dict:
-        packages_objects = dict()
-        for package in self.document_object["packages"]:
-            packages_objects[package["SPDXID"]] = package
+    def update_dict_item_with_new_item(self, current_state: Dict, key: str, item_to_add: str) -> None:
+        if key not in current_state:
+            current_state[key] = [item_to_add]
+        elif item_to_add not in current_state:
+            current_state[key].append(item_to_add)
+
+    def create_relationships(self) -> List[Dict]:
+        packages_spdx_ids = [package.spdx_id for package in self.document.packages]
+        files_spdx_ids = [file.spdx_id for file in self.document.files]
+        packages_by_spdx_id = {package["SPDXID"]: package for package in self.document_object["packages"]}
```

Comment:
At this point we need the packages from `document_object` (which is the dict that is dumbed later on whereas `document` is the internal data model) as we are adding the `hasFiles` field to the already "written" packages dicts. So, we could use the packages from `document_object` when creating the list of spdx_ids in line 43. But this would imply that we use the literal string "SPDXID" to get the corresponding values and as the packages in `document` and `document_object` should be the same here, I would leave this. Is this okay for you?

Reply:
added a comment